pkg/plugins/plugins.go

```diff
@@ -130,10 +128,18 @@ func ReadPluginJSON(reader io.Reader) (JSONData, error) {
 	switch plugin.ID {
 	case "grafana-piechart-panel":
 		plugin.Name = "Pie Chart (old)"
-	case "grafana-pyroscope-datasource": // rebranding
-		plugin.Alias = "phlare"
-	case "debug": // panel plugin used for testing
-		plugin.Alias = "debugX"
+	case "grafana-pyroscope-datasource":
```

Comment:
Nit: can we move this switch-case to a map? I see this list growing and a switch-case is slower/more difficult to maintain.

Reply:
this whole section should be removed -- the only reason it exists is that we are waiting for parallel logic to exist in gcom.  We want to be careful that anyone is not able to publish a plugin with an alias that we do not notice until it is too late :)

Reply:
I actually like having a copy of the list here (even if there is a method in GCOM). That would help with the offline case.

Reply:
Can't we just say - we implement logic in gcom and we make sure we dont publish any plugin or plugin updates that we dont want to have an alias. Then we can (here in the core) trust that the ones that are published and have an alias in plugin.json are good to have an alias? Otherwise it will become hard to keep those hardcoded lists in sync.

Reply:
A (small) problem with that would be an unsigned local plugin, which could impersonate the legit version of the plugin. Not sure if we need to worry about that.

Reply:
@tolzhabayev -- exactly.  I have the check here so we can get everything working here and get it working in GCOM in parallel.

Reply:
GCOM Ticket https://github.com/grafana/grafana-com/issues/7467 to double check we have the same approach in mind